server/app/controllers/Application.scala

```diff
@@ -3,13 +3,19 @@ package controllers
 import com.google.inject.Inject
 import controllers.Application._
 import controllers.UserController._
+import dal.Dao
+import dal.Dao._
+import models.TaskType
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.mvc._
 
-class Application @Inject()(val messagesApi: MessagesApi) extends Controller with I18nSupport {
+import scala.concurrent.ExecutionContext
 
-  def index = Action { implicit request =>
-    Ok(views.html.index())
+class Application @Inject()(dao: Dao, val messagesApi: MessagesApi)(implicit ec: ExecutionContext) extends Controller with I18nSupport {
+
+  def index = Action.async { implicit request =>
+    val tasks = dao.getTasks(TaskType.scalaClass, 20, now)
```

Comment:
No, our DAO never throw exceptiona, check this. But we have Future here, so we need to .recover it